src/handlers/gestures/GestureDetector.tsx

```diff
@@ -145,14 +162,16 @@ function attachHandlers({
     RNGestureHandlerModule.attachGestureHandler(
       gesture.handlerTag,
       viewTag,
-      !useAnimated // send direct events when using animatedGesture, device events otherwise
+      !gesture.shouldUseReanimated // send direct events when using animatedGesture, device events otherwise
     );
   }
 
   if (preparedGesture.animatedHandlers) {
-    preparedGesture.animatedHandlers.value = (gesture.map(
-      (g) => g.handlers
-    ) as unknown) as HandlerCallbacks<Record<string, unknown>>[];
+    preparedGesture.animatedHandlers.value = (gesture
+      .filter((g) => g.shouldUseReanimated) // ignore gestures that shouldn't run on UI
```

Comment:
Instead of comment you can just declare `const isAnimatedGesture = (g) => g.shouldUseReanimated`.

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/1862/commits/f6af3008e67af7e85deb4ca81858a282c093bada#diff-b3c286a84982c75e6aef5f631088fc65b125a290a7391298c2c5c41cddff32e3R175.